tools/run_tests/run_interop_tests.py

```diff
@@ -777,7 +777,7 @@ def cloud_to_prod_jobspec(language,
     ]
     if transport_security == 'tls':
         transport_security_options = ['--use_tls=true']
-    elif transport_security == 'google_default_credentials' and language == 'c++':
+    elif transport_security == 'google_default_credentials' and (language == 'c++' or language == 'go'):
         transport_security_options = [
             '--custom_credentials_type=google_default_credentials'
         ]
```

Comment:
I am taking a closer look at how the `JobSpec` suite name is being formed on line 820, and it looks to me like both the google default credentials and TLS prod to cloud test cases will have the same "shortname" (this would be a pre-existing problem). I think this might be breaking something in how reporting is set up (https://github.com/grpc/grpc/blob/master/tools/run_tests/python_utils/jobset.py#L461)